lms/djangoapps/discussion_api/api.py

```diff
@@ -110,6 +110,27 @@ def _is_user_author_or_privileged(cc_content, context):
     )
 
 
+def _update_vote_count(voted, serializer):
+    """
+    Updates the vote_count in the given serializer
+
+    If the user has already voted and `voted == True` do nothing,
```

Comment:
How can you tell whether the user has actually voted previously or not?  Are you basing it on what the client has passed in the API?  If so, wouldn't this be susceptible to malice?  That is, someone can keep sending upvotes even if they have already voted in the past.